verilog/formatting/formatter_test.cc

```diff
@@ -224,6 +224,132 @@ static const std::initializer_list<FormatterTestCase> kFormatterTestCases = {
+    {// macro call with MacroArg tokens as arugments and with semicolon
+     "`FOOOOOO(\nbar1...\n,\nbar2...\n,\nbar3...\n,\nbar4\n);\n",
+     "`FOOOOOO(bar1..., bar2..., bar3...,\n"
+     "         bar4);\n"},
+    {// macro declaration exceeds line length limit and contains semicolon
+     "`F_MACRO(looooooong_type if_it_fits_I_sits);\n",
+     "`F_MACRO(\n"
+     "    looooooong_type if_it_fits_I_sits);\n"},
+    {// macro call with not fitting arguments and semicolon
+     "`MACRO_FFFFFFFFFFF("
+            "type_a_aaaa,type_b_bbbbb,"
+            "type_c_cccccc,type_d_dddddddd,"
+            "type_e_eeeeeeee,type_f_ffff);\n",
+     "`MACRO_FFFFFFFFFFF(\n"
+     "    type_a_aaaa, type_b_bbbbb,\n"
+     "    type_c_cccccc, type_d_dddddddd,\n"
+     "    type_e_eeeeeeee, type_f_ffff);\n"},
+    {// nested macro call with semicolon
+     "`MACRO_FFFFFFFFFFF( "
+     "`A(type_a_aaaa), `B(type_b_bbbbb), "
+     "`C(type_c_cccccc), `D(type_d_dddddddd), "
+     "`E(type_e_eeeeeeee), `F(type_f_ffff));\n",
+     "`MACRO_FFFFFFFFFFF(`A(type_a_aaaa),\n"
+     "                   `B(type_b_bbbbb),\n"
+     "                   `C(type_c_cccccc),\n"
+     "                   `D(type_d_dddddddd),\n"
+     "                   `E(type_e_eeeeeeee),\n"
+     "                   `F(type_f_ffff));\n"},
+    {// two-level nested macro call with semicolon
+     "`MACRO_FFFFFFFFFFF( "
+     "`A(type_a_aaaa, `B(type_b_bbbbb)), "
+     "`C(type_c_cccccc, `D(type_d_dddddddd)), "
+     "`E(type_e_eeeeeeee, `F(type_f_ffff)));\n",
+     "`MACRO_FFFFFFFFFFF(\n"
+     "    `A(type_a_aaaa, `B(type_b_bbbbb)),\n"
+     "    `C(type_c_cccccc,\n"
+     "       `D(type_d_dddddddd)),\n"
+     "    `E(type_e_eeeeeeee,\n"
+     "       `F(type_f_ffff)));\n"},
+    {// three-level nested macro call with semicolon
+     "`MACRO_FFFFFFFFFFF(`A(type_a_aaaa,"
+     "`B(type_b_bbbbb,`C(type_c_cccccc))),"
+     "`D(type_d_dddddddd,`E(type_e_eeeeeeee,"
+     "`F(type_f_ffff))));\n",
+     "`MACRO_FFFFFFFFFFF(\n"
+     "    `A(type_a_aaaa,\n"
+     "        `B(type_b_bbbbb,\n"
+     "           `C(type_c_cccccc))),\n"
+     "    `D(type_d_dddddddd,\n"
+     "        `E(type_e_eeeeeeee,\n"
+     "           `F(type_f_ffff))));\n"},
+    {// macro call with no args
+     "`FOOOOOO()\n",
+     "`FOOOOOO()\n"},
+    {// macro call with no args and semicolon
+     "`FOOOOOO();\n",
+     "`FOOOOOO();\n"},
+    {// macro call with no args and semicolon separated by space
+     "`FOOOOOO() ;\n",
+     "`FOOOOOO();\n"},
+    {// macro call with comments in argument list
+     "`FOO(aa, //aa\nbb , // bb\ncc)\n",
+     "`FOO(aa,  //aa\n"
+     "     bb,  // bb\n"
+     "     cc)\n"},
+    {// macro call with comment before first argument
+     "`FOO(//aa\naa, //bb\nbb , // cc\ncc)\n",
+     "`FOO(  //aa\n"
+     "    aa,  //bb\n"
+     "    bb,  // cc\n"
+     "    cc)\n"},
+    {// macro call with argument including comment
+     "`FOO(aa, bb//cc\ndd)\n",
+     "`FOO(aa, bb//cc\n"
+     "dd)\n"},
```

Comment:
Should `dd)` be wrapped in +4 spaces?

Reply:
Second argument is parsed as single `MacroArg`:
```
        (#MacroArg @1104-1120: "//aa
bb // bb
cc")
```
and looks like rest of formatter is keeping its identation giving such result:
```
`FOO(aa, //aa
bb // bb
cc)
```

I think that problem is somewhere in `VerilogAnalyzer`, e.g. https://github.com/google/verible/blob/master/verilog/analysis/verilog_analyzer.cc#L128

Can you confirm that?

Reply:
Ah, I think I see it: attempting to parse the macro arg fails due to missing comma, if you put a comma after `bb`, it will then see two separate macro arguments, instead of `bb//cc\ndd` as one atomic unit.

Reply:
Yes (there is also test for that). I think that that is not related to #225 issue and should not be addressed in this PR.

I also think that this is not an expected behaviour and I'm guessing that it should be handled in separate issue. Am I right?

Reply:
Falling back to an unlexed `MacroArg` when there is a parse-error is expected behavior.  If additional work is needed for //comments to format correctly inside macro arguments, it is ok to leave it for a separate PR (TODO(fangism)).  It looked like you handled it so I thought it was a matter of covering it in a test case.

Reply:
Okay. I'm leaving it for another issue/PR. I've added `TODO`s for both test cases